common/src/main/java/net/adoptopenjdk/icedteaweb/ProcessUtils.java

```diff
@@ -69,4 +74,70 @@ public static void waitForSafely(final Process process) {
             }
         }
     }
+
+    /**
+     * Actively ignore the content of stdout and stderr.
+     * This happens in a separate thread as on windows the process will not terminate if a buffer is full.
+     * Reading in a separate thread ensures that the buffer is constantly drained.
+     *
+     * @param process the process to drain stdout and stderr
+     */
+    public static void ignoreStdOutAndStdErr(final Process process) {
+        ignoreStream(process.getInputStream());
+        ignoreStream(process.getErrorStream());
+    }
+
+    private static void ignoreStream(final InputStream in) {
+        new Thread(() -> {
+            try {
+                final byte[] buffer = new byte[1024];
+                while (in.read(buffer) > 0) {
+                    // do nothing;
+                }
+            } catch (IOException e) {
+                LOG.error(IcedTeaWebConstants.DEFAULT_ERROR_MESSAGE, e);
+                throw new RuntimeException(e);
+            }
+        }).start();
+    }
+
+    /**
+     * Reads the content of the stdout into a future string.
+     * This happens in a separate thread as on windows the process will not terminate if the out buffer is full.
+     * Reading in a separate thread ensures that the buffer is constantly drained.
+     * The future will complete when the process has exited.
+     *
+     * @param process the process to read the stdout from
+     * @return a future holding the content of the stdout
+     */
+    public static Future<String> readStdOutAsUtf8(final Process process) {
+        return readStreamAsUtf8(process.getInputStream());
+    }
+
+    /**
+     * Reads the content of the stderr into a future string.
+     * This happens in a separate thread as on windows the process will not terminate if the err buffer is full.
+     * Reading in a separate thread ensures that the buffer is constantly drained.
+     * The future will complete when the process has exited.
+     *
+     * @param process the process to read the stderr from
+     * @return a future holding the content of the stderr
+     */
+    public static Future<String> readStdErrAsUtf8(final Process process) {
+        return readStreamAsUtf8(process.getErrorStream());
+    }
+
+    private static Future<String> readStreamAsUtf8(final InputStream in) {
+        final CompletableFuture<String> result = new CompletableFuture<>();
+        new Thread(() -> {
```

Comment:
Please use executors (like Executors.newChachedThreadPool) instead of creating a new thread al the time